### PR TITLE
Upgrade eslint and inherit from airbnb

### PR DESCRIPTION
**Pull request by rylanc**
The results of this are minor.  I upgraded eslint and started inheriting from airbnb's config so we can slowly remove the rules we define. I override the current differences between our config and theirs. Here's the PR for ZP: https://github.com/Gusto/zenpayroll/pull/14560

### CODE REVIEW

File: base.js [hunk 1]
```diff
@@ -1,125 +1,45 @@
...
-  extends: 'eslint:recommended',
+  extends: 'airbnb-base',
   parser: 'babel-eslint',
   env: {
-    es6: true
+    es6: true,
   },
   parserOptions: {
-    ecmaVersion: 6
+    ecmaVersion: 6,
```
Comment (jeffreykwan) [inline base.js]: aww i understand this `,` will make diffs easier to read but i personally think it looks a bit ugly haha. :woman_shrugging:

**Comment by jeffreykwan** [comment]
@zp-alfred accept

**Comment by zp-alfred** [comment]
Accepted by jeffreykwan!

File: base.js (hunk 1 above)
Comment (rylanc) [inline base.js]: I actually agree, but I'd rather have one well-researched and thought out source of rules rather than debate and tweak own own config

**Pull request merged**